libvirt.go

```diff
@@ -159,82 +169,46 @@ func (d *Driver) validateNetwork() error {
 		return err
 	}
 	network, err := conn.LookupNetworkByName(d.Network)
-	if err == nil {
-		xmldoc, err := network.GetXMLDesc(0)
-		if err != nil {
-			return err
-		}
-		/* XML structure:
-		<network>
-		    ...
-		    <ip address='a.b.c.d' prefix='24'>
-		        <dhcp>
-		            <host mac='' name='' ip=''/>
-		        </dhcp>
-		*/
-		type Ip struct {
-			Address string `xml:"address,attr"`
-			Netmask string `xml:"prefix,attr"`
-		}
-		type Network struct {
-			Ip Ip `xml:"ip"`
-		}
-
-		var nw Network
-		err = xml.Unmarshal([]byte(xmldoc), &nw)
-		if err != nil {
-			return err
-		}
-
-		if nw.Ip.Address == "" {
-			return fmt.Errorf("%s network doesn't have DHCP configured properly", d.Network)
-		}
-		// Corner case, but might happen...
-		if active, err := network.IsActive(); !active {
-			log.Debugf("Reactivating network: %s", err)
-			err = network.Create()
-			if err != nil {
-				log.Warnf("Failed to Start network: %s", err)
-				return err
-			}
-		}
-		return nil
+	if err != nil {
+		return fmt.Errorf("Use 'crc setup' for define the network, %+v", err)
 	}
-	// TODO - try a couple pre-defined networks and look for conflicts before
-	//        settling on one
-
-
-	log.Debugf("Defining network...")
-	tmpl, err := template.New("network").Parse(crclibvirt.NetworkTemplate)
+	xmldoc, err := network.GetXMLDesc(0)
 	if err != nil {
 		return err
 	}
-
-	config := crclibvirt.NetworkConfig{
-		DomainName: d.MachineName,
-		MAC:        crclibvirt.MACAddress,
-		IP:         crclibvirt.IPAddress,
+	/* XML structure:
+	<network>
+	    ...
+	    <ip address='a.b.c.d' prefix='24'>
+	        <dhcp>
+	            <host mac='' name='' ip=''/>
+	        </dhcp>
+	*/
+	type Ip struct {
+		Address string `xml:"address,attr"`
+		Netmask string `xml:"prefix,attr"`
+	}
+	type Network struct {
+		Ip Ip `xml:"ip"`
 	}
 
-	var xml bytes.Buffer
-	err = tmpl.Execute(&xml, config)
+	var nw Network
+	err = xml.Unmarshal([]byte(xmldoc), &nw)
 	if err != nil {
 		return err
 	}
 
-	network, err = conn.NetworkDefineXML(xml.String())
-	if err != nil {
-		log.Errorf("Failed to create network: %s", err)
-		return nil
-	}
-	err = network.SetAutostart(true)
-	if err != nil {
-		log.Warnf("Failed to set network to autostart: %s", err)
+	if nw.Ip.Address == "" {
+		return fmt.Errorf("%s network doesn't have DHCP configured properly", d.Network)
```

Comment:
drop `properly` or 'properly configured'. however, the sentence is still kinda 'faux-detailed'.